beacon-chain/sync/pending_attestations_queue.go

```diff
@@ -261,62 +296,49 @@ func (s *Service) savePendingAtt(att ethpb.SignedAggregateAttAndProof) {
 	_, ok := s.blkRootToPendingAtts[root]
 	if !ok {
 		pendingAttCount.Inc()
-		s.blkRootToPendingAtts[root] = []ethpb.SignedAggregateAttAndProof{att}
+		s.blkRootToPendingAtts[root] = []any{pending}
 		return
 	}
-	// Skip if the attestation from the same aggregator already exists in
+
+	// Skip if the attestation/aggregate from the same validator already exists in
 	// the pending queue.
 	for _, a := range s.blkRootToPendingAtts[root] {
-		if attsAreEqual(att, a) {
+		if isEqual(a) {
 			return
 		}
 	}
+
 	pendingAttCount.Inc()
-	s.blkRootToPendingAtts[root] = append(s.blkRootToPendingAtts[root], att)
+	s.blkRootToPendingAtts[root] = append(s.blkRootToPendingAtts[root], pending)
 }
 
-func attsAreEqual(a, b ethpb.SignedAggregateAttAndProof) bool {
+func pendingAggregatesAreEqual(a, b ethpb.SignedAggregateAttAndProof) bool {
 	if a.Version() != b.Version() {
 		return false
 	}
+	return a.AggregateAttestationAndProof().GetAggregatorIndex() == b.AggregateAttestationAndProof().GetAggregatorIndex()
```

Comment:
why wouldn't we also check the other fields?

Reply:
You're right, we should check more